rover_gui/src/QTopUtilityBar/QTopUtilityBar.hpp

```diff
@@ -0,0 +1,100 @@
+#ifndef QTOP_UTILITY_BAR
+#define QTOP_UTILITY_BAR
+
+#include "rclcpp/rclcpp.hpp"
+#include "rover_lib2/helpers/ip_pinging.hpp"
+#include "rover_msgs/msg/gps.hpp"
+#include "rover_lib2/helpers/watchdog.hpp"
+
+#include "rclcpp/duration.hpp"
+#include "UI_TopUtilityBar.h"
+#include <QtWidgets/QGridLayout>
+#include <QtWidgets/QWidget>
+#include <cstddef>
+#include <rover_msgs/msg/battery.hpp>
+#include <rover_msgs/msg/gps.hpp>
+#include <rover_msgs/msg/wifi_connection.hpp>
+#include <QDateTime>
+#include <QTimeZone>
+
+class QTopUtilityBar : public QWidget
+{
+    Q_OBJECT
+
+    static constexpr const char* TOPIC_BATTERY = "/rover/auxiliary/battery";
+    static constexpr const char* TOPIC_WIFI_CONNECTION = "/rover/auxiliary/connection_speed";
+    static constexpr const char* TOPIC_GNSS = "/rover/gps/position";
+
+    static constexpr const size_t DELAY_CHECK_BATTERY_PUB_COUNT_MS = 1000UL;
+    static constexpr const size_t DELAY_CHECK_RSSI_PUB_COUNT_MS = 1000UL;
+    static constexpr const size_t DELAY_CHECK_GNSS_PUB_COUNT_MS = 1000UL;
+
+    static constexpr const size_t DELAY_UPDATE_TIMER_MS = 1000UL;
+
+    static constexpr const size_t WATCH_DOG_DELAY_MS = 1000UL;
+    static constexpr const float WATCH_DOG_TIMEOUT = 1.0f;
```

Comment:
Just to conform to new codebase standards
```suggestion
    static constexpr const float WATCH_DOG_TIMEOUT = 1.0F;
```

Reply:
done